### PR TITLE
Netapp: rename _facts -> _info

### LINKED ISSUE #60524: Netapp _facts modules need to be renamed to _info and stop returning ansible_facts
##### SUMMARY
According to the [requirements](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#contributing-to-ansible-objective-requirements), `_facts` modules should return information that is specific to the host the module runs on as `ansible_facts`. (See also ansible/ansible#54280 for some discussion.)

I think that the following modules do not satisfy this condition: they return information that is not specific to the host machine, but bound to an online account. Please verify whether this is actually the case.

To fix this, the core team thinks the best approach is to rename the `_facts` modules to `_info` modules, and change the return value so that if the module is called with the `_facts` name (should be deprecated, and will then be removed in Ansible 2.13), it should return `ansible_facts` as now (and print a deprecation message), and if it is called with the `_info` name, it no longer returns `ansible_facts`, so that the result needs to be registered. This informs users that they have to change something, won't break their playbooks, and allows them (by switching to the `_info` names) to verify that their roles and playbooks no longer use the facts.

If that approach doesn't work, or you want to change the module's interface (like rename the return value(s)), it is better to deprecate the `_facts` module and create a new `_info` module next to it. To see both approaches implemented, you can take a look at https://github.com/ansible/ansible/commit/2d98734ad51c3de557c3a7d1f7166bdd114ceb0f (ansible/ansible#60178).

If you have any input — preferred method to solve this, doubts that all modules should be renamed, ... — please write it here!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
na_ontap_gather_facts
~~netapp_e_facts~~

##### ANSIBLE VERSION
```paste below
2.9.0
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #60524.

##### ISSUE TYPE
- Bugfix Pull Request
- New Module Pull Request

##### COMPONENT NAME
na_ontap_gather_facts
netapp_e_facts

### CODE REVIEW

**Review by lonico — APPROVED** [review]
I'm OK with the change in na_ontap_info, but cannot talk about e_info.

**Comment by felixfontein** [comment]
@hulquest @ndswartz @lmprice @amit0701 @schmots1 @carchi8py can someone of you check `netapp_e_info`?

**Comment by thedoubl3j** [comment]
@ndswartz can you chime in on this?

**Comment by netapp-eseries** [comment]
What is the reason for the change?

**Comment by ndswartz** [comment]
What is the reason for the name change?

**Comment by felixfontein** [comment]
@ndswartz please check out #60524.

**Comment by ndswartz** [comment]
@felixfontein From my understanding everything gathered in the netapp_e_facts is information specific to the storage array. What information is bound to an online account?

**Comment by ndswartz** [comment]
I just noticed that the documentation has not been updated (I'm going to update it now). Currently, there are two ways to interact without storage, santricity web service proxy and santricity web services embedded. Either path produces the same in storage system specific information. Would it be preferable to change it in light of this information?

**Comment by felixfontein** [comment]
@ndswartz as I understand it, these modules connect to a service over the network (i.e. in general not to something running on the local machine). Is that correct? In which sense should the information returned be assumed to be specific to `inventory_hostname`? `ansible_facts` should only be used for something which is a fact about the host in `inventory_hostname`.

**Comment by ndswartz** [comment]
@felixfontein That's correct. For security purposes we interact directly with the storage system's rest api; however the information returned would be specific to the inventory_hostname.

**Comment by felixfontein** [comment]
@ndswartz does this appy to `netapp_e_facts`, or also to `na_ontap_gather_facts`?

**Comment by ndswartz** [comment]
@felixfontein Just netapp_e_facts

**Comment by felixfontein** [comment]
Ok, then I'll remove it from this PR.

File: lib/ansible/modules/storage/netapp/na_ontap_info.py [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+                "qos_adaptive_policy_info", "qos_policy_info", "security_key_manager_key_info",
+                "security_login_account_info", "storage_failover_info", "volume_info",
+                "vserver_info", "vserver_login_banner_info", "vserver_motd_info", "vserver_nfs_info"
+                Can specify a list of values to include a larger subset.  Values can also be used
+                with an initial C(M(!)) to specify that a specific subset should
+                not be collected.
+            - nvme is supported with ONTAP 9.4 onwards.
+            - use "help" to get a list of supported information for your system.
+        default: "all"
+        version_added: 2.8
```
Comment (samccann) [inline lib/ansible/modules/storage/netapp/na_ontap_info.py]:
```suggestion
        version_added: 2.9
```

Comment (felixfontein) [reply lib/ansible/modules/storage/netapp/na_ontap_info.py]: Thanks. Also removed it.

**Comment by felixfontein** [comment]
@lonico any second thoughts / change requests?

**Review by carchi8py — APPROVED** [review]

**Review by lonico — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
@lonico @thedoubl3j @ndswartz @samccann @carchi8py thanks all for discussing and reviewing this!

**Pull request merged**